Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+
+        Parameters
+        ----------
+        domain : Domain | None
+            If None, the domain is inferred from the data. Otherwise, specifies
+            the column assignment to the new Table.
+        X : np.ndarray
+            The X component of the data (or undetermined, depending on the domain).
+        Y : np.ndarray, optional, default None
+            The Y component of the data.
+        metas : np.ndarray, optional, default None
+            The meta attributes of the data.
+        weights : anything TableBase.set_weights accepts, optional, default None
+            The weights to use for the resulting Table.
+        Returns
+        -------
+        TableBase
+            A new Table constructed from the given data.
+        """
+        def correct_shape(what):
+            if what is None or len(what.shape) == 2:
+                return what
+            else:
+                return np.atleast_2d(what).T
+
+        # if anything is sparse, use the sparse version
+        from .impl import SparseTable
+        if sp.issparse(X) or (Y is not None and sp.issparse(Y)) \
+                          or (metas is not None and sp.issparse(metas)) \
+                          or (weights is not None and sp.issparse(weights))\
+                          or issubclass(cls, SparseTable):
+            # explicitly construct a sparse table as this can be called from Table(...),
+            # but allow subclasses to call correctly
+            if issubclass(cls, SparseTable):
+                return cls._from_sparse_numpy(domain, X, Y, metas, weights)
+            else:
+                return SparseTable._from_sparse_numpy(domain, X, Y, metas, weights)
+
+        if domain is None:
+            result = cls._from_data_inferred(X, Y, metas)
+            if weights is not None:
+                result.set_weights(weights)
+            return result
+
+        # ensure correct shapes (but not sizes) so we can iterate
+        X = correct_shape(X)
+        Y = correct_shape(Y)
+        metas = correct_shape(metas)
+
+        result = cls()
+        for role_array, variables in zip((X, Y, metas),
+                                         (domain.attributes, domain.class_vars, domain.metas)):
+            if role_array is None:
+                if variables:
+                    raise ValueError("Variable and column count mismatch. ")
+                continue
+            if role_array.shape[1] != len(variables):
+                raise ValueError("Variable and column count mismatch. ")
+            for column, variable in zip(role_array.T, variables):
+                result[variable.name] = column
+        result.domain = domain
+        if weights is not None:
+            result.set_weights(weights)
+
+        # transform any values we believe are null into actual null values
+        result.replace(to_replace=list(Variable.MISSING_VALUES), value=np.nan, inplace=True)
```

Comment:
No, because someone would want to construct a table with a list that contains "?" or a ndarray that contains "?" or a dataframe that contains "?" or similar. This doesn't only happen in io, but I do think it mostly happens in tests.